universal_transfer_operator/src/universal_transfer_operator/data_providers/base.py

```diff
@@ -57,11 +67,11 @@ def check_if_transfer_supported(self, source_dataset: DatasetType) -> bool:
         source_connection_type = get_dataset_connection_type(source_dataset)
         return Location(source_connection_type) in self.transfer_mapping
 
-    def read(self):
+    def read(self) -> Iterator[pd.DataFrame] | Iterator[FileStream]:
         """Read the dataset and write to local reference location"""
```

Comment:
We could improve this description. Do we read the dataset potentially from any data source (e.g. object store) and copy it to a local file system?